### PR TITLE
add accesspoints playbook

### PR DESCRIPTION
**Pull request by equinox0815**
this is still work in progress..

### CODE REVIEW

**Comment by equinox0815** [comment]
@nbraud waht to think about doing the global network settings like done here?

File: ansible/group_vars/accesspoints/main.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      options:
+        enabled: '1'
+        enable_server: '0'
+        server:
+          - '0.lede.pool.ntp.org'
+          - '1.lede.pool.ntp.org'
+          - '2.lede.pool.ntp.org'
+          - '3.lede.pool.ntp.org'
+
+  network:
```
Comment (nbraud) [inline ansible/group_vars/accesspoints/main.yml]: Shouldn't we templatize that from the `net` object?

File: ansible/group_vars/openwrt/main.yml [hunk 2]
```diff
@@ -8,7 +8,6 @@ openwrt_packages_remove:
 openwrt_packages_add:
   - haveged
   - htop
-  - hwclock
```
Comment (nbraud) [inline ansible/group_vars/openwrt/main.yml]: ~Why remove `hwclock`?~ I guess the APs do not have hardware clocks?

We can keep `hwclock` in the default, and put `"-hwclock"` in the accesspoint's extras.
OTOH, I'm not sure which option is the right default there.

File: ansible/roles/openwrt-image/defaults/main.yml [hunk 3]
```diff
@@ -2,7 +2,7 @@
 openwrt_variant: lede
 openwrt_release: 17.01.4
 openwrt_download_dir: "{{ global_cache_dir }}/openwrt"
-openwrt_tarball_basename: "{{ openwrt_variant }}-imagebuilder-{{ openwrt_release }}-{{ openwrt_arch }}{% if openwrt_target != 'generic' %}-{{ openwrt_target }}{% endif %}.Linux-x86_64"
+openwrt_tarball_basename: "{{ openwrt_variant }}-imagebuilder-{{ openwrt_release }}-{{ openwrt_arch }}-{{ openwrt_target }}.Linux-x86_64"
```
Comment (nbraud) [inline ansible/roles/openwrt-image/defaults/main.yml]: Just checked, works.

File: ansible/roles/openwrt-image/tasks/prepare.yml [hunk 4]
```diff
@@ -13,7 +13,7 @@
     state: directory
     mode: '0755'
   with_items:
-    - "{{ openwrt_download_dir }}/dl"
+    - "{{ openwrt_download_dir }}/dl/{{ openwrt_arch }}"
```
Comment (nbraud) [inline ansible/roles/openwrt-image/tasks/prepare.yml]: Nice  :+1:

**Comment by nbraud** [comment]
@equinox0815 The `net` stuff looks quite nifty.

File: ansible/roles/openwrt-image/tasks/prepare.yml (hunk 4 above)
Comment (equinox0815) [reply ansible/roles/openwrt-image/tasks/prepare.yml]: ok so i will move on with it. As you already mentioned above it doesn't hold all the info we want in there after all.

File: ansible/group_vars/openwrt/main.yml (hunk 2 above)
Comment (equinox0815) [reply ansible/group_vars/openwrt/main.yml]: I think most routers and other stuff that usually runs OpenWRT don't have hardware clocks and if they have one they are mostly not battery backed. Also given the fact that hardware clocks in Linux are only ever used/updated during boot and shutdown it doesn't make sense for most hardware to have hwclock support at all.

File: ansible/group_vars/accesspoints/main.yml (hunk 1 above)
Comment (equinox0815) [reply ansible/group_vars/accesspoints/main.yml]: yes sure, i just didn't get to doing it

**Comment by equinox0815** [comment]
ok so i will move on with the net stuff. As you already mentioned above it doesn't hold all the info we want in there after all.

File: ansible/group_vars/openwrt/main.yml (hunk 2 above)
Comment (nbraud) [reply ansible/group_vars/openwrt/main.yml]: Yeah, fair.

**Comment by equinox0815** [comment]
@nbraud i fixed the issue i was having and as you can see the solution is actually quite easy.... don't know why it took me so long to realize that this is the obvious solution.
The setup is already quite nice but i also want to enable 802.11r. Of course not without testing it first. I got 4 UAP AC-Lite which we can use. I hope i will have time to do this next week - preferably together with you...
Anyway as soon as we successfully tested our setup we can finally merge the accesspoint role and deploy it onto all the APs.

File: ansible/group_vars/accesspoints/main.yml [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+      country: AT
+      path: "{{ accesspoint_wireless_device_paths['2.4g'] }}"
+      htmode: 'HT20'
+
+
+## TODO: set up 802.11r see:
+##        * https://www.reddit.com/r/openwrt/comments/515oea/finally_got_80211r_roaming_working/
+##        * https://gist.github.com/lg/998d3e908d547bd9972a6bb604df377b
+accesspoint_wireless_ifaces: "{{ accesspoint_wireless_ifaces_yaml | from_yaml }}"
+accesspoint_wireless_ifaces_yaml: |
```
[the remark this answers is not in the document]
Comment (equinox0815) [reply ansible/group_vars/accesspoints/main.yml]: done.

File: ansible/group_vars/accesspoints/main.yml [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+      color_scheme=0
+      delay=15
+      left_meters=AllCPUs Memory Swap
+      left_meter_modes=1 1 1
+      right_meters=Tasks LoadAverage Uptime
+      right_meter_modes=2 2 2
+
+
+openwrt_uci:
+  system:
```
[the remark this answers is not in the document]
Comment (equinox0815) [reply ansible/group_vars/accesspoints/main.yml]: as we just discussed this we will do this refactoring in another PR

**Pull request merged**